creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -62,45 +81,89 @@ class SyncHistoryAggregator {
+    }
+
+    try {
+      const { success, fail } = SyncHistoryAggregator.getAggregateSyncKeys()
 
-    let successCount = await redisClient.get(success)
-    let failCount = await redisClient.get(fail)
+      let successfulSyncsCount = await redisClient.get(success)
+      let failedSyncsCount = await redisClient.get(fail)
 
-    successCount = successCount ? parseInt(successCount) : 0
-    failCount = failCount ? parseInt(failCount) : 0
+      successfulSyncsCount = successfulSyncsCount ? parseInt(successfulSyncsCount) : 0
+      failedSyncsCount = failedSyncsCount ? parseInt(failedSyncsCount) : 0
 
-    const currentAggregateData = {
-      success: successCount,
-      fail: failCount,
-      triggered: successCount + failCount
+      currentAggregateData.success = successfulSyncsCount
+      currentAggregateData.fail = failedSyncsCount
+      currentAggregateData.triggered = successfulSyncsCount + failedSyncsCount
+    } catch (e) {
+      logger.error(`syncHistoryAggregator - getAggregateSyncData() error - ${e.toString()}`)
     }
 
     // Structure: {triggered: <number>, success: <number>, fail: <number>}
     return currentAggregateData
   }
 
-  static async getLatestSyncData () {
-    const { success, fail } = SyncHistoryAggregator.getLatestSyncKeys()
+  /**
+   * Returns the date of the latest successful and failed sync. Will be `null` if a sync with those
+   * states have not been triggered.
+   * @param {Object} logContext the log context off of the Express req object
+   * @returns an object of the current day's aggregate sync count like
+   *     {success: <MM:DD:YYYYTHH:MM:SS:sssZ>, fail: <MM:DD:YYYYTHH:MM:SS:sssZ>}
+   */
+  static async getLatestSyncData (logContext) {
+    const logger = genericLogger.child(logContext)
+    let currentLatestSyncData = {
+      success: null,
+      fail: null
+    }
 
-    const successDate = await redisClient.get(success)
-    const failDate = await redisClient.get(fail)
+    try {
+      const { success, fail } = SyncHistoryAggregator.getLatestSyncKeys()
 
-    const currentLatestSyncData = {
-      success: successDate,
-      fail: failDate
-    }
+      const latestSyncSuccessTimestamp = await redisClient.get(success)
+      const latestSyncFailTimestamp = await redisClient.get(fail)
 
-    // Structure: {success: <latest date>, fail: <latest date>}
+      currentLatestSyncData.success = latestSyncSuccessTimestamp
+      currentLatestSyncData.fail = latestSyncFailTimestamp
+    } catch (e) {
+      logger.error(`syncHistoryAggregator - getLatestSyncData() error - ${e.toString()}`)
+    }
+    // Structure: {success: <MM:DD:YYYYTHH:MM:SS:sssZ>, fail: <MM:DD:YYYYTHH:MM:SS:sssZ>}
     return currentLatestSyncData
   }
 
+  /**
+   * Retrieves the redis keys used for storing aggregate sync counts
+   * @returns an object of the `success` and `fail` redis keys for the aggregate sync count
+   */
   static getAggregateSyncKeys () {
     // ex: aggregateSync:::05212021:::success
```

Comment:
this example should actually be `aggregateSync:::2021-05-28:::success` like the examples you added above, and smae for line 182 below

Reply:
good catch, updating